### PR TITLE
Swift: clean up `VarDecl`, `NamedPattern` and `SwitchStmt` interactions

### PR DESCRIPTION
**Pull request by redsun82**
* `variables` under `CaseStmt` are now AST children, which solves orphan `VarDecl`s in that case
* reordered `CaseStmt` AST children to be `labels > variables > body` (was `body > labels`)
* made `NamedPattern::getVarDecl` an extracted property instead of `getName`

The above was introducing several `VarDecl` duplications, as `CaseStmt::getCaseBodyVariables`, `NamedPattern::getVarDecl` and `DeclRefExpr::getDecl` were all actually returning distinct `VarDecl` instances. It turns out, `VarDecl` has a [`getCanonicalVarDecl()`](https://github.com/apple/swift/blob/96828e646a0e1d49505be8c0dacf01e63b9b6e40/include/swift/AST/Decl.h#L5764-L5779) to get the true unique variable declaration :shrug: This is now applied at a fundamental level in the the dispatcher, so that any fetching of a variable declaration done anywhere in any translator will pass through this.

### CODE REVIEW

File: swift/ql/test/extractor-tests/generated/decl/ConcreteVarDecl/ConcreteVarDecl_getParentPattern.expected [hunk 1]
```diff
@@ -1,3 +1,4 @@
+| var_decls.swift:2:7:2:7 | i | var_decls.swift:2:7:2:7 | i |
```
Comment (redsun82) [inline swift/ql/test/extractor-tests/generated/decl/ConcreteVarDecl/ConcreteVarDecl_getParentPattern.expected:1]: it's actually weird that this was missing! This is the `for` loop `i` variable, which is distinct from the `i` variable introduced after the loop.

Comment (redsun82) [reply swift/ql/test/extractor-tests/generated/decl/ConcreteVarDecl/ConcreteVarDecl_getParentPattern.expected:1]: ah, wait, this is a parent pattern thing, I don't fully know what to make of this :laughing:

File: swift/extractor/translators/StmtTranslator.cpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ codeql::ForEachStmt StmtTranslator::translateForEachStmt(const swift::ForEachStm
   auto entry = dispatcher.createEntry(stmt);
   fillLabeledStmt(stmt, entry);
   entry.body = dispatcher.fetchLabel(stmt.getBody());
- // entry.sequence = dispatcher.fetchLabel(stmt.getTypeCheckedSequence());
+  // entry.sequence = dispatcher.fetchLabel(stmt.getTypeCheckedSequence());
```
Comment (MathiasVP) [inline swift/extractor/translators/StmtTranslator.cpp:76]: Why not just delete this line?

Comment (redsun82) [reply swift/extractor/translators/StmtTranslator.cpp:76]: this was autofromatted by by my IDE, I have no idea what it is :sweat_smile:

Comment (redsun82) [reply swift/extractor/translators/StmtTranslator.cpp]: yeah, I removed it, it was an unneeded leftover :+1:

**Review by redsun82 — COMMENTED** [review]
@MathiasVP the DB scripts look good to me!

**Comment by MathiasVP** [comment]
Awesome. In that case I'll approve the merge so that we can merge it 🎉 (because the _code_ changes LGTM as well!)

**Review by MathiasVP — APPROVED** [review]

File: swift/ql/test/library-tests/ast/PrintAst.expected [hunk 3]
```diff
@@ -6651,62 +6643,24 @@ patterns.swift:
...
-#   85| [ConcreteVarDecl] a
-#   85|     Type = Int
-#   85| [ConcreteVarDecl] b
-#   85|     Type = Int
-#   88| [ConcreteVarDecl] a
-#   88|     Type = Int
 #   92| [ConcreteVarDecl] x
 #   92|     Type = (Int)
 #   95| [ConcreteVarDecl] x
 #   95|     Type = Int
```
Comment (redsun82) [inline swift/ql/test/library-tests/ast/PrintAst.expected:6743]: still quite a number of orphaned (or maybe duplicate?) var decls here, eh?

Comment (MathiasVP) [reply swift/ql/test/library-tests/ast/PrintAst.expected:6743]: Looks like it, yeah

**Pull request merged**